src/server/connection.rs

```diff
@@ -3114,6 +3142,42 @@ impl Connection {
         self.inner.send(msg.into());
         self.supported_encoding_flag = (true, not_use);
     }
+
+    #[cfg(not(any(target_os = "android", target_os = "ios")))]
+    async fn handle_window_focus(&mut self) {
+        if let Ok(displays) = super::display_service::update_get_sync_displays().await {
+            let current_display = crate::get_focused_display(displays);
+            if let Some(idx) = current_display {
+                let mut pi = PeerInfo::default();
+                pi.current_display = idx as i32;
+                let mut msg_out = Message::new();
+                msg_out.set_peer_info(pi);
+                self.send(msg_out).await;
+            }
+        }
+    }
+
+    #[cfg(not(any(target_os = "android", target_os = "ios")))]
+    async fn handle_cursor_switch_display(&mut self, pos: CursorPosition) {
+        if let Ok(displays) = super::display_service::update_get_sync_displays().await {
```

Comment:
Considering that it runs every 33 milliseconds, please test whether the cached display info in display_service.rs or the sent display information is sufficient for the execution.